File: Beecrowd-Python/1118.py
# Loop principal
while True:
    # Inicializa variáveis
    notas_validas = 0
    soma_notas = 0

    # Loop para obter duas notas válidas
    while notas_validas < 2:
        nota = float(input())

        # Verifica se a nota está no intervalo de 0 a 10
        if nota < 0 or nota > 10:
            print("nota invalida")
        else:
            notas_validas += 1
            soma_notas += nota

    # Calcula a média das duas notas
    media = soma_notas / 2
    print("media = {:.2f}".format(media))

    # Loop para verificar se o usuário deseja realizar um novo cálculo
    while True:
        print("novo calculo (1-sim 2-nao)")
        x = int(input())

        # Se o usuário escolhe 1, sai do loop interno e realiza um novo cálculo
        if x == 1:
            break
        # Se o usuário escolhe 2, encerra o programa
        elif x == 2:
            exit()

File: Beecrowd-C/1118.c
#include <stdio.h>
#include <stdlib.h>

int main() {
    // Declaração de variáveis
    int resposta = 0;
    float nota1, nota2;

    // Loop principal que continua até que o usuário escolha sair (resposta == 2)
    while (resposta != 2) {
        // Solicita ao usuário a inserção da primeira nota
        scanf("%f", &nota1);

        // Validação da primeira nota para garantir que esteja no intervalo de 0 a 10
        while (nota1 < 0 || nota1 > 10) {
            printf("nota invalida\n");
            scanf("%f", &nota1);
        }

        // Solicita ao usuário a inserção da segunda nota
        scanf("%f", &nota2);

        // Validação da segunda nota para garantir que esteja no intervalo de 0 a 10
        while (nota2 < 0 || nota2 > 10) {
            printf("nota invalida\n");
            scanf("%f", &nota2);
        }

        // Calcula a média das duas notas e imprime o resultado
        printf("media = %.2f\n", (nota1 + nota2) / 2);

        // Loop para obter a resposta do usuário (1-sim, 2-não) e validar a entrada
        do {
            printf("novo calculo (1-sim 2-nao)\n");
            scanf("%d", &resposta);
        } while (resposta < 1 || resposta > 2);
    }

    return 0;
}